ql/time/schedule.cpp

```diff
@@ -463,7 +463,7 @@ namespace QuantLib {
                    result.dates_[0]);
         if (truncationDate<result.dates_.back()) {
             // remove later dates
-            while (result.dates_.back()>truncationDate) {
```

Comment:
Unfortunately, `Schedule` grew a lot of different cases.  It might happen that `result.isRegular_` is empty but `result.dates_` is not.  In that case, we should leave `isRegular` alone but still pop from the dates until we reach the truncation date.

Reply:
Can you please provide a concrete example? I was referring to [one of your blogs](https://www.implementingquantlib.com/2014/11/odds-and-ends-date-calculations.html). If isRegular's length is not dates.size() - 1, then what will isRegular's meaning become?

Reply:
If the schedule builds the dates, isRegular will be populated and its size will be dates.size()-1.
However, [another constructor](https://github.com/lballabio/QuantLib/blob/master/ql/time/schedule.hpp#L46) allows passing a pre-built vector of dates; isRegular might also be passed (in which case [the same constraint apply](https://github.com/lballabio/QuantLib/blob/master/ql/time/schedule.cpp#L93)) or it might be left empty. In that case, we don't know if the dates are regular or not.

Reply:
What does an empty isRegular mean if dates.size >= 2? 

Still, to make it work, we may only need to add a if( !isRugular_.empty()) before popping it inside the while loop block.

Reply:
It means we don't know if the dates are regular.  Thanks, this works now.